### PR TITLE
Add CHANGELOG

### PR DESCRIPTION
**Pull request by SergioGasquez**
- Add changelog
- Add CI check to enforce changelog modifications

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -104,3 +104,19 @@ jobs:
...
+
+  changelog:
+    name: Changelog
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+
+      - name: Check that changelog updated
+        uses: dangoslen/changelog-enforcer@v3
```
Comment (jessebraham) [inline .github/workflows/rust.yml:116]: This action drives me crazy, it does not work consistently with forks. I'm fine with enforcing the `CHANGELOG` but I would prefer to use something else.

Comment (SergioGasquez) [reply .github/workflows/rust.yml:116]: I've only found https://github.com/Zomzog/changelog-checker but it seems like is not actively developed (it still uses NodeJS12 and has a issue open for it with no action taken so far) so I will remove the check for the time being.

Comment (jessebraham) [reply .github/workflows/rust.yml]: Okay, thanks for checking at least.

**Review by jessebraham — APPROVED** [review]
LGTM, thanks again

**Pull request merged**